packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -1083,20 +1083,15 @@ class RenderFlex extends RenderBox with ContainerRenderObjectMixin<RenderBox, Fl
     if (size.isEmpty)
       return;
 
-    if (clipBehavior == Clip.none) {
-      _clipRectLayer.layer = null;
-      defaultPaint(context, offset);
-    } else {
-      // We have overflow and the clipBehavior isn't none. Clip it.
-      _clipRectLayer.layer = context.pushClipRect(
-        needsCompositing,
-        offset,
-        Offset.zero & size,
-        defaultPaint,
-        clipBehavior: clipBehavior,
-        oldLayer: _clipRectLayer.layer,
-      );
-    }
+    // We have overflow and the clipBehavior isn't none. Clip it.
```

Comment:
How do we know that the behavior isn't none here?

Reply:
Ah, this refers to the old tests. pushClipRect protects against clipRect none. I'd delete the comment or modify it to fit the new conditions.

Reply:
this is out of date, my bad.

Reply:
fixed